content/en/blog/_posts/2022-04-07-pipy-as-k8s-infra-container-for-service-registration/index.md

```diff
@@ -0,0 +1,205 @@
+---
+layout: blog
+title: "Pipy as Kubernetes Infra container - Service Registration"
+date: 
+slug: pipy-as-k8s-infra-container-for-service-registration
+---
+
+**Author:** Zhang Xiaohui (Addo), Ali Naqvi
+
+The infrastructure or “pause” container exists in every Pod in Kubernetes. When Pod is scheduled to a node, kubelet on the node creates an operating environment for it, including namespace, network and IPC, before creating a container. This environment is called the Pod Sandbox, and the infra container is an important component and carrier of the Sandbox. It serves as the parent container for all containers in pod and serves as the basis of shared namespace, ipc, network, shared volume etc.
+
+Kubelet first creates the infra container, and then other containers in Pod are added to the infra container's network namespace in the form of join namespace to achieve network communication between containers. Below is a depiction of a Pod in a service mesh environment:
+
+<img alt="Image of a pod infra container" src="kubernetes-pod-infra-arch.jpg">
+
+Kubernetes maintains a multi-architecture image that includes support for Windows. For Kubernetes v1.23 the recommended pause image is `k8s.gcr.io/pause:3.6`, which is very small and goes into a pause state once started.
+
+## Pipy as Infrastructure container
+
+With above information and details, we can see that in fact, infrastructure containers are invisible **sidecar**. Our exploration on `Pipy as Infra` series is two fold, one is to implement more infrastructure related functionality on this ubiquitous sidecar to make it a real infrastructure container, another one is to reduce the complexity and resource consumption associated with multiple sidecar containers.
```

Comment:
What's Pipy? At this point in the article, I don't know and I think most readers wouldn't. The other question readers will have is along the lines of “&hellip;and why should I care?” - it'd be good to answer both.